wkcuber/cubing.py

```diff
@@ -156,8 +170,48 @@ def cubing_job(args):
                     ]
 
                 buffer = prepare_slices_for_wkw(slices, num_channels)
+                if downsampling_needed:
+                    logging.info(f"Downsampling buffer of size {buffer.shape}.")
+                    padding_size_for_downsampling = [
+                        (
+                            0,
+                            (
+                                required_multiple_of
+                                - (buffer_dimension_size % required_multiple_of)
+                            )
+                            % required_multiple_of,
+                        )
+                        for (required_multiple_of, buffer_dimension_size) in zip(
+                            target_mag.to_array(), buffer.shape[1:]
+                        )
+                    ]
+                    logging.info([(0, 0)] + padding_size_for_downsampling)
+                    buffer = np.pad(
+                        buffer,
+                        pad_width=[(0, 0)] + padding_size_for_downsampling,
+                        mode="constant",
+                    )
+                    downsampled_buffer_shape = [
+                        current_dimension_size // dimension_decrease
```

Comment:
Similarly, if both these variables are numpy arrays, `downsampled_buffer_shape = current_dimension_size // dimension_decrease` is enough for this expression. You might need to convert back from np to python lists.

Reply:
Very nice :wink: